### PR TITLE
Add softmax_backward to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/softmax_backward.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                           xla::int64 dim) {
+  auto lower_for_shape_fn =
+      [dim](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK_EQ(operands.size(), 2)
+        << "Unexpected number of operands: " << operands.size();
+    return BuildSoftmaxGrad(/*grad_output=*/operands[0],
+                            /*output=*/operands[1], dim);
+  };
+  return InferOutputShape({grad_output.shape(), output.shape()},
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/softmax_backward.cpp]: Do we really need to infer it?

Comment (asuhan) [reply torch_xla/csrc/ops/softmax_backward.cpp]: No we don't, fixed the silliness in LogSoftmaxBackward as well.

File: torch_xla/csrc/softmax_builder.cpp [hunk 2]
```diff
@@ -46,6 +46,19 @@ SoftMaxPartials LogSoftmaxPartials(const xla::XlaOp& logits, xla::int64 dim) {
   return {std::move(broadcast_dimensions), shifted_logits, exp_shifted, reduce};
 }
 
+xla::XlaOp SoftmaxSumOfGrad(const xla::XlaOp& grad_output, xla::int64 dim) {
+  xla::Shape grad_output_shape = XlaHelpers::ShapeOfXlaOp(grad_output);
+  auto broadcast_dimensions =
+      BroadcastDimensions(grad_output_shape.rank(), dim);
+  xla::XlaBuilder* builder = grad_output.builder();
```
Comment (dlibenzi) [inline torch_xla/csrc/softmax_builder.cpp]: Why caching the builder?

File: torch_xla/csrc/tensor_methods.cpp [hunk 3]
```diff
@@ -1432,6 +1433,14 @@ XLATensor XLATensor::softmax(const XLATensor& input, xla::int64 dim) {
       XlaHelpers::GetCanonicalDimensionIndex(dim, input.shape().get().rank())));
 }
 
+XLATensor XLATensor::softmax_backward(const XLATensor& grad_output,
+                                      const XLATensor& output, xla::int64 dim) {
+  return grad_output.CreateFrom(ir::MakeNode<ir::ops::SoftmaxBackward>(
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: This is right about at the boundary, for as far as nested, single line ops, should be.

File: torch_xla/csrc/softmax_builder.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/softmax_builder.cpp]: Done.

File: torch_xla/csrc/tensor_methods.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Done.

**Pull request merged**